src/python/pants/backend/python/typecheck/mypy/subsystem.py

```diff
@@ -39,6 +40,14 @@ def register_options(cls, register):
             advanced=True,
             help="Path to `mypy.ini` or alternative MyPy config file",
         )
+        register(
+            "--junit-xml-dir",
```

Comment:
MyPy has support for lots of report types, including HTML: https://mypy.readthedocs.io/en/stable/config_file.html#report-generation

I was going to propose making this more generic like `--report-dir`, and then having the user set the relevant options in their `mypy.ini`. But I don't think that would work - we need to know what file path they specified so that we can capture it.

I think, then, we'd end up with multiple different options, like `--html-report-dir`? Or maybe like the Coverage plugin, have `--report-dir` and `--report-type` as two separate options? Thoughts?

It's fine to not implement this yet, only want to make sure we have a path forward.